service/service.go

```diff
@@ -466,27 +480,53 @@ func (t *service) AddRecord(ctx context.Context, id thread.ID, body format.Node)
 		return
 	}
 
+	// Update head
+	if err = t.store.SetHead(id, lg.ID, rec.Cid()); err != nil {
+		return nil, err
+	}
+
 	log.Debugf("added record %s (thread=%s, log=%s)", rec.Cid().String(), id, lg.ID)
 
 	// Notify local listeners
-	r = &record{
-		Record:   rec,
-		threadID: id,
-		logID:    lg.ID,
-	}
-	if err = t.bus.SendWithTimeout(r, time.Second*3); err != nil {
+	r = NewRecord(rec, id, lg.ID)
+	if err = t.bus.SendWithTimeout(r, notifyTimeout); err != nil {
```

Comment:
Every time I see the `SendWithTimeout` thing, something feels bad. If some receiver is too slow to read these things it will make this kind of important operations fail. From the subscriber POV, subscribing to something doesn't seem like a super important thing to handle well... and just a single slow receiver might block all operations.

I think ultimately is a problem of the bus API... I think there should be subscribers which can't miss any event in the bus and make the original operation fail (which is a quite aggressive policy, like in the Store since it can't miss an event), and maybe others that should be ignored if can't be pushed the new notification (but not make the original operation fail). Not sure would be a real solution, just first thought.

Not something we should fix now... but this thing seems to be waiting to bite us.